src/interface/IVault.sol

```diff
@@ -5,15 +5,16 @@ import {IERC4626} from "src/Common.sol";
 
 interface IVault is IERC4626 {
     struct VaultStorage {
-        bool paused;
         uint256 totalAssets;
         address provider;
         address buffer;
+        bool paused;
+        uint8 decimals;
```

Comment:
the vault can have a different number of decimals from the underlying primary deposit asset.

This way we can for example have 18 decimals, but still have USDC as a primary deposit asset.